gpytorch/mlls/variational_elbo.py

```diff
@@ -1,91 +1,77 @@
 #!/usr/bin/env python3
 
-import torch
-from .marginal_log_likelihood import MarginalLogLikelihood
-from .. import settings
-
-
-class VariationalELBO(MarginalLogLikelihood):
-    def __init__(self, likelihood, model, num_data, combine_terms=True):
-        """
-        A special MLL designed for variational inference
-        Args:
-        - likelihood: (Likelihood) - the likelihood for the model
-        - model: (Module) - the variational GP model
-        - num_data: (int) - the total number of training data points (necessary for SGD)
-        - combine_terms: (bool) - whether or not to sum the expected NLL with the KL terms (default True)
-        """
-        super(VariationalELBO, self).__init__(likelihood, model)
-        self.combine_terms = combine_terms
-        self.num_data = num_data
+from ._approximate_mll import _ApproximateMarginalLogLikelihood
+
+
+class VariationalELBO(_ApproximateMarginalLogLikelihood):
+    r"""
+    The variational evidence lower bound (ELBO). This is used to optimize
+    variational Gaussian processes (with or without stochastic optimization).
+
+    .. math::
+
+       \begin{align*}
+          \mathcal{L}_\text{ELBO} &=
+          \mathbb{E}_{p_\text{data}( y, \mathbf x )} \left[
+            \log \mathbb{E}_{q(\mathbf u)} \left[  p( y \! \mid \! \mathbf u) \right]
+          \right] - \beta \: \text{KL} \left[ q( \mathbf u) \Vert p( \mathbf u) \right]
+          \\
+          &\approx \sum_{i=1}^N \mathbb{E}_{q( \mathbf u)} \left[
+            \log \int p( y_i \! \mid \! f_i) p(f_i \! \mid \! \mathbf u) \: d \mathbf f_i
+          \right] - \beta \: \text{KL} \left[ q( \mathbf u) \Vert p( \mathbf u) \right]
+       \end{align*}
+
+    where :math:`N` is the number of datapoints, :math:`q(\mathbf u)` is the variational distribution for
+    the inducing function values, and `p(\mathbf u)` is the prior distribution for the inducing function
+    values.
+
+    :math:`\beta` is a scaling constant that reduces the regularization effect of the KL
+    divergence. Setting :math:`\beta=1` (default) results in the true variational ELBO.
+
+    For more information on this derivation, see `Scalable Variational Gaussian Process Classification`_
+    (Hensman et al., 2015).
+
+    :param ~gpytorch.likelihoods.Likelihood likelihood: The likelihood for the model
+    :param ~gpytorch.models.ApproximateGP model: The approximate GP model
+    :param int num_data: The total number of training data points (necessary for SGD)
+    :param float beta: (optional, default=1.) A multiplicative factor for the KL divergence term.
+        Setting it to 1 (default) recovers true variational inference
+        (as derived in `Scalable Variational Gaussian Process Classification`_).
+        Setting it to anything less than 1 reduces the regularization effect of the model
+        (similarly to what was proposed in `the beta-VAE paper`_).
+    :param bool combine_terms: (default=True): Whether or not to sum the
```

Comment:
when beta != 1 and combine_terms=False is the returned KL scaled?

Reply:
With the current implementation, the KL is always scaled (even when combine_terms=False). Open to changing this though.

Reply:
I'm in favor of keeping it as is. If people want to do something more complicated with beta they should set beta=1. and do whatever they want.